src/OSPSuite.Core/Domain/Formulas/TableFormulaWithXArgument.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System.Collections.Generic;
+using OSPSuite.Core.Domain.Services;
+
+namespace OSPSuite.Core.Domain.Formulas
+{
+   public class TableFormulaWithXArgument : Formula
+   {
+      public string TableObjectAlias { get; set; }
+      public string XArgumentAlias { get; set; }
+
+      /// <summary>
+      ///    Add path to the object with table formula
+      /// </summary>
+      /// <param name="tableObjectPath"></param>
+      public void AddTableObjectPath(IFormulaUsablePath tableObjectPath)
+      {
+         TableObjectAlias = tableObjectPath.Alias;
+         AddObjectPath(tableObjectPath);
+      }
+
+      /// <summary>
+      ///    Add path to the object with offset formula
+      /// </summary>
+      /// <param name="xArgumentObjectPath"></param>
+      public void AddXArgumentObjectPath(IFormulaUsablePath xArgumentObjectPath)
+      {
+         XArgumentAlias = xArgumentObjectPath.Alias;
+         AddObjectPath(xArgumentObjectPath);
+      }
+
+      /// <summary>
+      ///    Returns table formula object
+      /// </summary>
+      /// <param name="refObject">Entity using formula</param>
+      public IUsingFormula GetTableObject(IUsingFormula refObject)
+      {
+         return GetReferencedEntityByAlias(TableObjectAlias, refObject) as IUsingFormula;
+      }
+
+      /// <summary>
+      ///    Returns reference object used for table formula
+      /// </summary>
+      /// <param name="refObject">Entity using formula</param>
+      public IFormulaUsable GetXArgumentObject(IUsingFormula refObject)
+      {
+         return GetReferencedEntityByAlias(XArgumentAlias, refObject);
+      }
+
+      /// <summary>
+      ///    Return the value of the table object for 0-offset value
+      /// </summary>
+      protected override double CalculateFor(IEnumerable<IObjectReference> usedObjects, IUsingFormula dependentObject)
+      {
+         var tableObject = GetTableObject(dependentObject);
+         var tableFormula = tableObject?.Formula as TableFormula;
+
+         return tableFormula?.ValueAt(GetXArgumentObject(dependentObject).Value) ?? double.NaN;
+      }
```

Comment:
return NaN or better throw an Exception? (TableWithArgFormula uses a non-Table object)

Reply:
I am in favor of NaN but I can change if you prefer an exception

Reply:
I am in favor of NaN but I can change if you prefer an exception